_drafts/ember-orbit.md

```diff
@@ -0,0 +1,477 @@
+---
+title: Offline Data with Ember-Orbit
+---
+
+[Orbit](http://orbitjs.com/) is a powerful library for data management, and [Ember-Orbit](https://github.com/orbitjs/ember-orbit) integrates it nicely with Ember. It allows you to set up sophisticated data flows among multiple sources--but how these work can be hard to understand when you're getting started. Let’s walk through setting up Ember-Orbit in a tiny app, to give us access to both read and write data while offline, then sync up when we come back online. Almost all the steps in this app follow what's in the official Orbit and Ember-Orbit getting started guides, but we'll also see concrete examples of what functionality each step adds to the app.
+
+The example app we’ll build is the start of a messaging app. We'll build a feature allowing us to send a new message and show it in a list:
+
+(SCREENSHOT)
+
+
+## Setting Up the Backend
+
+The backend for our app will be a Rails API, [offline-api](https://github.com/CodingItWrong/offline-api). It’s set up following the pattern of ember-orbit’s [peeps-uuids example](https://github.com/cerebris/peeps-uuids). Follow the instructions in the readme to download and set it up. Run the API using the command `bin/rails s`.
+
+Note that instead of using autoincrement integers for the primary keys, this API uses UUIDs assigned by the frontend. Orbit is set up to handle IDs this way by default. This allows records to be created locally even when the backend isn’t accessible.
+
+## Installing Orbit
+
+To get started on the frontend, create a new Ember app:
+
+```sh
+# ember new offline-ember
+```
+
+Install the `ember-orbit` addon:
+
+```sh
+# ember install ember-orbit
+```
+
```

Comment:
Also uninstall ember-data.

Reply:
I was hoping that was at least an option, glad to hear it.